paasta_tools/chronos_tools.py

```diff
@@ -420,12 +424,9 @@ def format_chronos_job_dict(self, docker_url, docker_volumes, dockercfg_location
         if self.get_schedule() is not None:
             complete_config['schedule'] = self.get_schedule()
         else:
-            matching_parent_pairs = [(parent, get_job_for_service_instance(*parent.split(".")))
-                                     for parent in self.get_parents()]
-            for parent_pair in matching_parent_pairs:
-                if parent_pair[1] is None:
-                    raise InvalidParentError("%s has no matching jobs in Chronos" % parent_pair[0])
-            complete_config['parents'] = [parent_pair[1]['name'] for parent_pair in matching_parent_pairs]
+            # The input to parents is the normal paasta syntax, but for chronos we have to
+            # convert it to what chronos expects, which uses its own spacer
+            complete_config["parents"] = map(paasta_to_chronos_job_name, self.get_parents())
```

Comment:
Hah. Don't like map? :)

Yea I knew that this would make it so we "fail early" if the parents don't exist.
I thought this was "ok" as they would eventually get there and it wasn't quite worth the effort to kinda "look into the future."

Maybe as an optimization we can make `list_chronos_jobs` emit things parent-first to minimize early failures?